src_common/common_linux.inl

```diff
@@ -1162,6 +1162,27 @@ String get_executable_path()
     return make_string(path);
 }
 
+String get_current_working_directory()
+{
+    umm buffer_size = 1024;
+    while (buffer_size < Kilobyte(10))
```

Comment:
Linux defines PATH_MAX in its build, which limits the length of strings some APIs can return. The underlying filesystem may not impose any limits, so the 10KB limitation is technically incorrect. getcwd() is a replacement for getwd() with that limitation, but on Linux there is additionally getcwd(NULL, 0) which returns a malloced buffer.

Best usage would probably be trying to use getcwd() on a buffer of length PATH_MAX on stack, and if that fails, call getcwd(NULL, 0). Then in either case copy that string to temp, instead of wasting a minimum of 1024 bytes on the path.

Additionally, if the process is running under a different filesystem root (eg. by using chroot, or by running in a privileged mount space), the cwd might not even be reachable if it's above the root. In this case getcwd() returns a string prefixed with "(unreachable)”, and which should be reported as an error.